routing/road_graph.hpp

```diff
@@ -19,22 +20,29 @@ class Junction
 {
 public:
   Junction();
-  Junction(m2::PointD const & point);
+  Junction(m2::PointD const & point, feature::TAltitude altitude);
   Junction(Junction const &) = default;
   Junction & operator=(Junction const &) = default;
 
   inline bool operator==(Junction const & r) const { return m_point == r.m_point; }
   inline bool operator<(Junction const & r) const { return m_point < r.m_point; }
 
   inline m2::PointD const & GetPoint() const { return m_point; }
+  inline feature::TAltitude GetAltitude() const { return m_altitude; }
 
 private:
   friend string DebugPrint(Junction const & r);
 
   // Point of the junction
   m2::PointD m_point;
+  feature::TAltitude m_altitude;
 };
 
+inline Junction MakeJunctionForTesting(m2::PointD const & point)
+{
+  return Junction(point, feature::kInvalidAltitude);
```

Comment:
There's situations when one end of a feature segment has a valid altitude and the other one  has invalid altitude. Such features are considered as feature without altitude. If I use zero instead of kInvalidAltitude there will be big and wrong altitude difference on some features.